zeroconf/__init__.py

```diff
@@ -2533,10 +2540,21 @@ def notify_all(self) -> None:
     def get_service_info(self, type_: str, name: str, timeout: int = 3000) -> Optional[ServiceInfo]:
         """Returns network's service information for a particular
         name and type, or None if no service matches by the timeout,
-        which defaults to 3 seconds."""
+        which defaults to 3 seconds.
+
+        Setting the timeout to 0 will avoid doing an I/O and only load
```

Comment:
I'm trying to think if I like special-casing `timeout` 0 like that. Have you considered adding a separate method for loading from cache? (like `get_service_info_from_cache(type_: str, name: str)`?)

Reply:
I think I can remove the change here since when I went to implement this in the places that consume the change I ended up writing it as creating the ServiceInfo object and then calling load_from_cache on it as I wanted to be very sure no I/O was happening in async context.

I'll need to write out a new set of tests for this. 

Will adjust later today if things calm down